### PR TITLE
Add plant initialization for SwimmingPool:Indoor.

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
This pull request includes a fix to a problem with swimming pools.  The problem was that flow was not being registered properly on plant loops being sized resulting in zero flow in certain situations even though the pool heating should have been running.  This has now been fixed by adding a plant call and separating some loop calls into a separate subroutine.  A new unit test was also added to test the new subroutine to make sure that the plant is being updated properly.  Tests with the defect input file show that the swimming pool flow is now properly accounted for in the results.  This issue is recorded in GitHub Issue #6763.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue #6763

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [x] Code style (parentheses padding, variable names)
 - [x] Functional code review (it has to work!)
 - [x] If defect, results of running current develop vs this branch should exhibit the fix
 - [x] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check -- verify this
 - [x] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add rules to spreadsheet
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Documentation changes in place
 - [ ] Changed docs build successfully
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

File: src/EnergyPlus/SwimmingPool.cc [hunk 1]
```diff
@@ -683,6 +683,7 @@ namespace SwimmingPool {
                 if (Pool(PoolNum).WaterInletNode > 0) {
                     InitComponentNodes(0.0, Pool(PoolNum).WaterMassFlowRateMax, Pool(PoolNum).WaterInletNode, Pool(PoolNum).WaterOutletNode,
                                        Pool(PoolNum).HWLoopNum, Pool(PoolNum).HWLoopSide, Pool(PoolNum).HWBranchNum, Pool(PoolNum).HWCompNum);
+                    PlantUtilities::RegisterPlantCompDesignFlow(Pool(PoolNum).WaterInletNode, Pool(PoolNum).WaterVolFlowMax);
```
Comment (rraustad) [inline src/EnergyPlus/SwimmingPool.cc]: This is inside a BeginEnvrnFlag test and is probably the wrong place. Other models use IF ( !SysSizingCalc), so this change is tentative.

**Comment by Myoldmopar** [comment]
@rraustad Could you verify the changes are still accurate, I resolved the conflict using Github's web resolver, so I didn't build the resulting commit locally...yet.

**Comment by rraustad** [comment]
This is correct. Thanks. No need to build, @RKStrand will take it from here.

**Comment by RKStrand** [comment]
@Myoldmopar I am taking this over and am currently working on it.  I am a little surprised that there are so many fails.  The fix is essentially the addition of one subroutine call and earlier (before bringing the code up to speed with develop) it seemed like the diffs were fairly minor and probably expected.  The fails in the latest ci run are thus surprising.  Is this another one of those cases where my GitHub desktop did something weird?

**Comment by Myoldmopar** [comment]
Don't sweat it for the moment.  I just deleted the weird result and we'll let it re-run.  The results from [nrel-bot-2](https://github.com/NREL/EnergyPlus/pull/6765#commitcomment-29412300) look more appropriate, with one file showing meaningful diffs.

**Comment by Myoldmopar** [comment]
Those results look better now.  Carry on...

**Comment by RKStrand** [comment]
@Myoldmopar Thanks for rerunning this.  Still not always 100% confident in the desktop app and what it does so I'm quick to suspect it when something goes wrong.  Glad it wasn't the issue here.

**Comment by Myoldmopar** [comment]
@RKStrand just a heads up this still needs a unit test before it can be accepted.  Still plenty of time in this release cycle before the bug fix cut off, just making sure it was on your radar.

**Comment by RKStrand** [comment]
@Myoldmopar Yup--I'm not done with it yet and it definitely needs a unit test.  It is what I am working on right now and will be doing a lot of bug fixes between now and the cut off.  So, if there is a bug fix that you know of that needs to be addressed and its looking for a fixer, feel free to send it my way.  Thanks!

**Comment by RKStrand** [comment]
@rraustad Okay, this has been implemented with a unit test in the latest commit.  So, this should be ready for review.  Would you be willing to review it?  If not, let me know and I'll point it to someone else.  Thanks!

**Review by rraustad — COMMENTED** [review]
All looks good here. Good cleanup, new function handles initialization and unit test exercises new function.

**Comment by rraustad** [comment]
Only diff's are in swimming pool example file where component sizes are now much larger to handle pool load.

**Comment by rraustad** [comment]
@RKStrand you have to modify the description to include what impact the changes have in case a user looks at this issue. Include a workaround if any. The changelog.html in the release folder will point to this pull request and a user will look here for information. So be clear on what changed and always include "This corrects issue #6763" to point to the issue and reason for this pull request.

**Comment by RKStrand** [comment]
@rraustad Wow--I totally fell down on the job on altering the description of the pull request.  Thanks for pointing that out.  I have made edits above so hopefully this is now ready to go.  Thanks again!

**Comment by rraustad** [comment]
Reviewed and diff's show increased plant capacity for swimming pool example file. Merging this in.

**Pull request merged**